model/osemosys.txt

```diff
@@ -577,281 +577,281 @@ solve;
+#                                                                                                                                         }
+#                                                                                                 }
+#                                                 }
+# #
+# ### Annual Emissions by Technology ###
+# #
+# printf "\n" >> ResultsPath & "/SelectedResults.csv";
+# printf "Annual Emissions by Technology (Emissions Units)" >> ResultsPath & "/SelectedResults.csv";
+# printf "\n" >> ResultsPath & "/SelectedResults.csv";
+# for {r in REGION} {printf ",%s", r >> ResultsPath & "/SelectedResults.csv";
+#                                         printf "\n" >> ResultsPath & "/SelectedResults.csv";
+#                                         for {t in TECHNOLOGY}         {printf "%s", t >> ResultsPath & "/SelectedResults.csv";
+#                                                                                         for {e in EMISSION}{printf",%s",e >> ResultsPath & "/SelectedResults.csv";
+#                                                                                                                         }
+#                                                                                         printf "\n" >> ResultsPath & "/SelectedResults.csv";
+#                                                                                         for {y in YEAR }         {
+#                                                                                                                                 printf "%g", y >> ResultsPath & "/SelectedResults.csv";
+#                                                                                                                                 for {e in EMISSION}{
+#                                                                                                                                                                 printf ",%g", sum{l in TIMESLICE, m in MODEperTECHNOLOGY[t]: EmissionActivityRatio[r,t,e,m,y]<>0} EmissionActivityRatio[r,t,e,m,y]*RateOfActivity[r,l,t,m,y]*YearSplit[l,y] >> ResultsPath & "/SelectedResults.csv";
+#                                                                                                                                                                 }
+#                                                                                                                                 printf "\n" >> ResultsPath & "/SelectedResults.csv";
+#                                                                                                                                 }
+#                                                 printf "\n" >> ResultsPath & "/SelectedResults.csv";
+#                                                                                         }
+#                                         }
 
 table AnnualEmissions
     {r in REGION, e in EMISSION, y in YEAR:
```

Comment:
Did you miss to to hash-out these last lines?

Reply:
No. otoole cant process glpk solution files. So we generate the solution file so snakemake is happy, but also generate a folder of CSVs directly via glpk. Since `SelectedResults.csv` does not follow otoole formatting, it was easier to just comment it out, as most people (I think?) dont really look at it haha. Maybe it should just be straight deleted though, instead of commented out.

Reply:
Ah, ok, then I had not looked very careful. I had thought you had hashed out all print statements apart from the last ones. Yes, removing the selectedResults makes sense.